trending_data/src/main/java/com/google/sps/servlets/Search.java

```diff
@@ -0,0 +1,107 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.sps.servlets;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.Video;
+import com.google.api.services.youtube.model.VideoListResponse;
+import com.google.sps.data.YTVid;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+public class Search {
+
+  /** Identify the developer's API key. */
+  private static final String apiKey =
+      "AIzaSyClpjaOk1vj_SEtnbkhFLQJya4ZoG7jq6c"; // key to access API
+
+  public static final HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+  public static final JsonFactory JSON_FACTORY = new JacksonFactory();
+  private static final long NUMBER_OF_VIDEOS_RETURNED = 6;
+
+  private static final ArrayList<YTVid> EMPTY_LIST = new ArrayList<>();
+
+  /** Initialize a YouTube object to search for videos on YouTube */
+  private static YouTube youtube;
+
+  /** Initialize a YouTube object to search for videos on YouTube */
+  public static ArrayList<YTVid> getData(String regionCode) {
+    Properties properties = new Properties();
+    try {
+      // Object to make YouTube Data API requests
+      youtube =
+          new YouTube.Builder(
+                  HTTP_TRANSPORT,
+                  JSON_FACTORY,
+                  new HttpRequestInitializer() {
+                    public void initialize(HttpRequest request) throws IOException {}
+                  })
+              .setApplicationName("search-by-location")
+              .build();
+
+      // Define the API request for retrieving result.
+      YouTube.Videos.List search = youtube.videos().list("snippet,contentDetails,statistics");
+
+      search.setKey(apiKey);
+      search.setMaxResults(NUMBER_OF_VIDEOS_RETURNED);
+      search.setRegionCode(regionCode);
+      search.setChart("mostPopular");
+
+      // Call the API and return results.
+      VideoListResponse searchResponse = search.execute();
+      List<Video> searchResultList = searchResponse.getItems();
+
+      if (searchResultList != null) {
```

Comment:
It's also possible that the request is successful but returns no results (as we saw for Egypt). I think in this case it would also be beneficial to return at least a log statement like for the errors below.